pkg/utils/resources/resources.go

```diff
@@ -93,20 +93,95 @@ func Subtract(lhs, rhs v1.ResourceList) v1.ResourceList {
 	return result
 }
 
-// Ceiling calculates the max between the sum of container resources and max of initContainers
-func Ceiling(pod *v1.Pod) v1.ResourceRequirements {
-	var resources v1.ResourceRequirements
+func PodRequests(pod *v1.Pod) v1.ResourceList {
+	reqs := v1.ResourceList{}
+
 	for _, container := range pod.Spec.Containers {
-		resources.Requests = MergeInto(resources.Requests, MergeResourceLimitsIntoRequests(container))
-		resources.Limits = MergeInto(resources.Limits, container.Resources.Limits)
+		containerReqs := container.Resources.Requests
+
+		MergeInto(reqs, containerReqs)
+
 	}
+
+	restartableInitContainerReqs := v1.ResourceList{}
+	initContainerReqs := v1.ResourceList{}
+
 	for _, container := range pod.Spec.InitContainers {
-		resources.Requests = MaxResources(resources.Requests, MergeResourceLimitsIntoRequests(container))
-		resources.Limits = MaxResources(resources.Limits, container.Resources.Limits)
+		containerReqs := container.Resources.Requests
```

Comment:
I'm confused by the logic below. What's the need to keep a bunch of temporary variables here? As I understand it, we should be taking the total of all container requests + the total of all initContainers with restart policy always + (the max resources of the remaining init containers). Based on this, I'm not sure that we have to keep track of the individual restartableInitContainerReqs and initContainerReqs, I think we can just merge this into the total requests

Reply:
> As I understand it, we should be taking the total of all container requests + the total of all initContainers with restart policy always + (the max resources of the remaining init containers)

Consider a pod with containers with these requests:
```
IC - 1 CPU
SC - 1 CPU
IC - 2 CPU

MC - 1 CPU
```

Your algorithm would give a total pod requests of 2 CPUs (1 from main container, 1 from the sidecar), which is incorrect.  The actual value is 3 CPUs (that sidecar is running, and then you need 2 CPUs for the init container).  If the order were changed:

```
IC - 1 CPU
IC - 2 CPU
SC - 1 CPU

MC - 1 CPU
```

then you're back to only needing 2 CPUs.

Reply:
Shoudn't it be like ? 
```Pod CPU request = Main container CPU request + Side container CPU request + max(Init container 1 CPU request, Init container 2 CPU request)

Pod CPU request = 1 + 1 + max(1, 2)

Pod CPU request = 4 CPUs 
```

Reply:
Yep, you're correct. I understood the algorithm incorrectly. Just discussed offline with @sadath-12 and we clarified on this.